src/VPolygon.jl

```diff
@@ -276,6 +277,123 @@ function ∈(x::AbstractVector{N}, P::VPolygon{N})::Bool where {N<:Real}
     return true
 end
 
+"""
+    rand(::Type{VPolygon}; [N]::Type{<:Real}=Float64, [dim]::Int=2,
+         [rng]::AbstractRNG=GLOBAL_RNG, [seed]::Union{Int, Nothing}=nothing
+        )::VPolygon{N}
+
+Create a random polygon in vertex representation.
+
+### Input
+
+- `VPolygon`     -- type for dispatch
+- `N`            -- (optional, default: `Float64`) numeric type
+- `dim`          -- (optional, default: 2) dimension
+- `rng`          -- (optional, default: `GLOBAL_RNG`) random number generator
+- `seed`         -- (optional, default: `nothing`) seed for reseeding
+- `num_vertices` -- (optional, default: `-1`) number of vertices of the
+                    polygon (see comment below)
+
+### Output
+
+A random polygon in vertex representation.
+
+### Algorithm
+
+We follow the idea [here](https://stackoverflow.com/a/47358689) based on
+[P. Valtr. Probability that n random points are in convex
+position](https://link.springer.com/article/10.1007%2FBF01271274).
+
+The number of vertices can be controlled with the argument `num_vertices`.
+For a negative value we choose a random number in the range `1:10`.
+"""
+function rand(::Type{VPolygon};
+              N::Type{<:Real}=Float64,
+              dim::Int=2,
+              rng::AbstractRNG=GLOBAL_RNG,
+              seed::Union{Int, Nothing}=nothing,
+              num_vertices::Int=-1
+             )::VPolygon{N}
+    @assert dim == 2 "a polygon must have dimension 2"
+    rng = reseed(rng, seed)
+    if num_vertices < 0
+        num_vertices = rand(1:10)
+    end
+
+    # special cases, 0 or 1 vertex
+    if num_vertices <= 1
+        vertices = [randn(rng, N, 2) for i in 1:num_vertices]
+        return VPolygon(vertices; apply_convex_hull=false)
+    end
+
+    # general case, >= 2 vertices
+
+    function remove_duplicates_sorted!(v::AbstractVector)
+        for i in length(v)-1:-1:1
+            if v[i] == v[i+1]
+                splice!(v, i+1)
+            end
+        end
+        return v
+    end
+
+    function random_axis_aligned_vectors(rng, N, n)
```

Comment:
can one substitute `random_axis_aligned_vectors` to picking n-elements randomly from a normal distribution? or what do these have in common?

```julia
julia> random_axis_aligned_vectors(rng, N, 5)
5-element Array{Float64,1}:
  0.12830892377309455
  2.387169565427545
  0.3232363202950286
 -0.29623066338265214
 -2.542484146113016

julia> random_axis_aligned_vectors(rng, N, 5)
5-element Array{Float64,1}:
  0.9638834005050456
  1.0636721009415822
  0.8336680989665766
 -1.5437334714482494
 -1.317490128964955

julia> random_axis_aligned_vectors(rng, N, 5)
5-element Array{Float64,1}:
  1.8457700493139375
  0.9765805620918393
 -0.40900877044843353
 -0.5103436430987645
 -1.9029981978585788
```

Reply:
The sum is zero. Have a look at the video [here](http://cglab.ca/~sander/misc/ConvexGeneration/convex.html).

Reply:
Very nice video.

I would suggest to attach a docstring and rename `random_axis_aligned_vectors`, maybe to `random_zero_sum_vector`.

Reply:
would this work?

```julia
function random_zero_sum_vector(rng, N, n)
    list = randn(rng, N, n)
    return [list[1:1:end-1] - list[2:1:end];
                list[end] - list[1]]
end
```

Reply:
Yes, but the numbers are not uniformly distributed. The article I linked to claims that the algorithm guarantees that, and I think it is good to have this property (if you can have it).

Reply:
alright, i think this function is interesting in its own right and i think we can put it outside (eventually call it `_random_zero_sum_vector`, we had this discussion 😄 ) and add docstring with reference to the article.